single_cell/workflows/cohort_qc/tasks.py

```diff
@@ -45,52 +44,91 @@ def generate_segmental_copynumber(hmmcopy_files, segmental_cn, sample):
     )
 
 
-def merge_mafs(germline_mafs, somatic_mafs, merged_maf):
+def format_maf_types(maf):
+    """Format maf.
+
+    Args:
+        maf ([pandas.DataFraame]): [maf data]
+
+    Returns:
+        [pandas.DataFrame]: [formatted maf]
+    """
+    float_cols = [
+        'AF', 'AFR_AF', 'AMR_AF', 'EAS_AF', 'EUR_AF', 'SAS_AF',
+        'AA_AF', 'EA_AF',
+        'ExAC_AF', 'ExAC_AF_AFR', 'ExAC_AF_AMR', 'ExAC_AF_EAS', 'ExAC_AF_FIN',
+        'ExAC_AF_NFE', 'ExAC_AF_OTH', 'ExAC_AF_SAS', 'vcf_qual', 'ExAC_AF_Adj',
+        'gnomAD_AF', 'gnomAD_AFR_AF', 'gnomAD_AMR_AF', 'gnomAD_ASJ_AF',
+        'gnomAD_EAS_AF', 'gnomAD_FIN_AF', 'gnomAD_NFE_AF', 'gnomAD_OTH_AF',
+        'gnomAD_SAS_AF', "MOTIF_SCORE_CHANGE"
+    ]
+
+    floatcols = maf.dtypes[maf.dtypes == "float64"]
+
+    for col in floatcols.index:
+        if col in float_cols:
+            pass
+        else:
+            maf[col] = maf[col].astype(pd.Int64Dtype())
+
+    return maf
+
+
+def merge_mafs(mafs, merged_maf):
+    """Write maf m to path merged_maf with label label (append).
+
+    Args:
+        mafs ([dict]): [sample: maf path]
+        merged_maf ([str]): [output path]
+
+    Yields:
+        [type]: [description]
+    """
     def _read_maf_chunk(maf_file, label):
         maf = pd.read_csv(maf_file, sep="\t", dtype='str', chunksize=1e7)
         for maf_chunk in maf:
+            maf_chunk = format_maf_types(maf_chunk)
```

Comment:
I'm reading the entire maf as str now

see dtype=str in the prev command.

so you dont need to ensure the dtypes. as long as its always read as str, all cols will stay the same and pandas wont add .0

Reply:
you probably dont need to type cast here anymore